addons/website_slides/tests/test_slide_utils.py

```diff
@@ -134,6 +136,33 @@ def test_mail_completed_with_different_templates(self):
             ['Congratulation! You completed %s' % self.channel.name, 'ATestSubject']
         )
 
+    def test_default_completion_time(self):
+        """Verify whether the system calculates the completion time when it is not specified,
+        but if the user does provide a completion time, the default time should not be applied."""
+
+        file_path = get_module_resource('base', 'tests', 'minimal.pdf')
+        pdf_content = Path(file_path).read_bytes()
```

Comment:
Hello @std-odoo 

The primary challenge in this task is that it consistently calculated the default time based on the number of pages present in the PDF document. Consequently, I found it necessary to include a PDF upload in the test case to ensure that it functions as intended, IMO 

Thanks :slightly_smiling_face:

Reply:
can't you use something different like a video, or writing manually the completion ?

Reply:
Hello @std-odoo 

No, i cannot use any other thing  because default time is only caluclate in the document type file and Video type only 
but the problem was only coming from the document type file only

in my fix ` _on_change_document_binary_content()` is getting called 

```
@api.onchange('document_binary_content')
    def _on_change_document_binary_content(self):
        if self.slide_category == 'document' and self.source_type == 'local_file' and self.document_binary_content:
            completion_time = self._get_completion_time_pdf(base64.b64decode(self.document_binary_content))
            if completion_time:
                self.completion_time = completion_time


```
in which `_get_completion_time_pdf()` is called which only calculate the time of the PDF only

So eventually I have to use a function to read the content of the file but it is giving the security error on runbot.

Thanks :slightly_smiling_face:

Reply:
or you can do that
```py
from unittest.mock import patch

def _get_completion_time_pdf(*args, **kwargs):
    return 13.37

with patch(
    'odoo.addons.website_slides.models.slide_slide.Slide._get_completion_time_pdf',
    new=_get_completion_time_pdf
):
    slides = self.env['slide.slide'].create({
        'name': 'Test_Content',
        'slide_category': 'document',
        'is_published': True,
        'is_preview': True,
        'document_binary_content': 'c3Rk',
        'channel_id': self.channel.id,
    })
self.assertEqual(13.37, slides.completion_time)
```